### PR TITLE
Docs deprecation warnings (SYN-4257)

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Fix use warnings.warn in synapse.common
- Add docs for enabling deprecation warnings.
- Fix use of List.length() in service.del implementation.
- Update agenda to create and use a view directly instead of calling core.eval.
- Generally remove Core.eval / CoreApi.eval from unit tests.
- Update function docstrings for deprecated functions to note their status.

### CODE REVIEW

File: docs/synapse/devopsguide.rst [hunk 1]
```diff
@@ -557,6 +557,38 @@ The Telepath URLs can be provided to the Storm API as follows::
     $lib.cell.trimNexsLog(consumers=$mirrors)
 
 
+.. _devops-deprecation-warnings:
+
+Viewing Deprecation Warnings
+----------------------------
+
+When functionality in Synapse is deprecated, it is marked with with the standard Python warnings_ mechanism to note
```
Comment (Cisphyx) [inline docs/synapse/devopsguide.rst:565]: 'with with'

File: synapse/lib/agenda.py [hunk 2]
```diff
@@ -870,7 +870,9 @@ async def _runJob(self, user, appt):
             success = False
             try:
                 opts = {'user': user.iden, 'view': appt.view, 'vars': {'auto': {'iden': appt.iden, 'type': 'cron'}}}
-                async for node in self.core.eval(appt.query, opts=opts):
+                opts = self.core._initStormOpts(opts)
```
Comment (Cisphyx) [inline synapse/lib/agenda.py:873]: i don't think this `_initStormOpts` is needed since 'user' is already populated

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -3812,7 +3812,7 @@ class SudoCmd(Cmd):
     name = 'sudo'
 
     async def execStormCmd(self, runt, genr):
-        s_common.deprecated('stormcmd:sudo')
+        s_common.deprecated(f'storm command: sudo')
```
Comment (Cisphyx) [inline synapse/lib/storm.py:3815]: doesn't need to be an f-string

Comment (Cisphyx) [inline synapse/lib/storm.py:3815]: doesn't need to be an f-string

File: synapse/lib/agenda.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/agenda.py:873]: I will double check.

Comment (vEpiphyte) [reply synapse/lib/agenda.py:873]: We could omit it; but there is a bit of a "if _initStormOpts does other things in the future, we'll miss out". It's an inexpensive call currently.

Comment (Cisphyx) [reply synapse/lib/agenda.py:873]: gotcha, makes sense

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**